PlayCover/Views/Settings/IPASourceSettings.swift

```diff
@@ -0,0 +1,263 @@
+        })
+        .buttonStyle(.plain)
+        .popover(isPresented: $showingPopover) {
+            Text(NSLocalizedString(popoverText, comment: ""))
+                .padding(10)
+        }
+    }
+}
+
+enum SourceValidation {
+    case valid, badurl, badjson, checking
+}
+
+struct AddSourceView: View {
+    @State var newSource = ""
+    @State var newSourceURL: URL?
+    @State var sourceValidationState = SourceValidation.checking
+    @Binding var addSourceSheet: Bool
+    @EnvironmentObject var storeVM: StoreVM
+
+    var body: some View {
+        VStack {
+            TextField(text: $newSource, label: {Text("preferences.textfield.url")})
+            Spacer()
+                .frame(height: 20)
+            HStack {
+                switch sourceValidationState {
+                case .valid:
+                    Image(systemName: "checkmark.circle.fill")
+                        .foregroundColor(.green)
+                    Text("preferences.popover.valid")
+                        .font(.system(.subheadline))
+                case .badurl:
+                    Image(systemName: "xmark.circle.fill")
+                        .foregroundColor(.red)
+                    Text("preferences.popover.badurl")
+                        .font(.system(.subheadline))
+                case .badjson:
+                    Image(systemName: "xmark.circle.fill")
+                        .foregroundColor(.red)
+                    Text("preferences.popover.badjson")
+                        .font(.system(.subheadline))
+                case .checking:
+                    EmptyView()
+                }
+                Spacer()
+                Button(action: {
+                    addSourceSheet.toggle()
+                }, label: {
+                    Text("button.Cancel")
+                })
+                Button(action: {
+                    if newSourceURL != nil {
+                        storeVM.appendSourceData(SourceData(source: newSourceURL!.absoluteString))
+                        addSourceSheet.toggle()
+                    }
+                }, label: {
+                    Text("button.OK")
+                })
+                .tint(.accentColor)
+                .keyboardShortcut(.defaultAction)
+                .disabled(sourceValidationState != .valid)
+            }
+        }
+        .padding()
+        .frame(width: 400, height: 100)
+        .onChange(of: newSource) { source in
+            validateSource(source)
+        }
+    }
+
+    func validateSource(_ source: String) {
+        sourceValidationState = .checking
+        DispatchQueue.global(qos: .userInteractive).async {
+            if let url = URL(string: source) {
+                newSourceURL = url
+                if StoreVM.checkAvaliability(url: newSourceURL!) {
+                    do {
+                        if newSourceURL!.scheme == nil {
+                            newSourceURL = URL(string: "https://" + newSourceURL!.absoluteString)!
```

Comment:
Local links are not being accepted as valid in the current scheme. `file:///path/to/my/apps/my_apps.json`

Reply:
Do we really need to support local links?